src/Interpreters/Context.cpp

```diff
@@ -3436,6 +3458,8 @@ ReadSettings Context::getReadSettings() const
     res.mmap_threshold = settings.min_bytes_to_use_mmap_io;
     res.priority = settings.read_priority;
 
+    res.throttler = getRemoteReadThrottler();
```

Comment:
May be worth to rename `throttler` in `ReadSettings` to something like `throttler_for_remote`? As it is always assigned `getRemoteReadThrottler()`